python/src/aiconfig/editor/server/server.py

```diff
@@ -59,7 +65,8 @@ def run_backend_server(edit_config: EditServerConfig) -> Result[str, str]:
             LOGGER.info("Initialized server state")
             debug = edit_config.server_mode in [ServerMode.DEBUG_BACKEND, ServerMode.DEBUG_SERVERS]
             LOGGER.info(f"Running in {edit_config.server_mode} mode")
-            app.run(port=edit_config.server_port, debug=debug, use_reloader=True)
+            use_reloader = edit_config.server_mode in [ServerMode.DEBUG_BACKEND, ServerMode.DEBUG_SERVERS]
```

Comment:
what does use_reloader mean in this context?

Reply:
> Enable automatic reloads on code changes.
--

https://pgjones.gitlab.io/hypercorn/how_to_guides/configuring.html


Unclear if it actually works tough

Reply:
>Unclear if it actually works tough

If its unclear if it works then either that should be tested or removed from the pr

Reply:
> Enable automatic reloads on code changes.

I'm missing the context on why server_mode is the option you pass into use_reloader. Could you explain what is going on here?

Reply:
3 things:

1. I mixed up this and V2 server which uses hypercorn. Reloader does work with flask (it doesn't seem to work with either uvicorn or hypercorn).

2. This change doesn't add the use of reloader, it just turns _off_ reloader if we're not in a debug mode. This change is clearly correct: it just changes the use_reloader condition based on server mode. 
  b) However, I just noticed that i already gave this condition a name, `debug` and should reuse that. 
3. The motivation for this logic is so you can reload the server on change if and only if you're debugging.